rollup/core/classes/RollupRepository.cls

```diff
@@ -48,15 +48,27 @@ public without sharing class RollupRepository {
     try {
       countAmount = Database.countQueryWithBinds(this.args.query, this.args.bindVars, this.accessLevel);
     } catch (Exception ex) {
-      RollupLogger.Instance.log('an error occurred while trying to get count query:\n' + this.args.query, ex, LoggingLevel.WARN);
+      RollupLogger.Instance.log('an error occurred while trying to get count query', ex, LoggingLevel.WARN);
       // not all count queries are valid, particularly those with polymorphic fields referencing parent fields
       // return a sentinel value instead, to be checked for downstream
       countAmount = SENTINEL_COUNT_VALUE;
     }
+    this.createQueryLog('Returned amount: ' + countAmount);
     return countAmount;
   }
 
   public override String toString() {
     return this.args.query + '\n' + 'with bind var keys: ' + this.args.bindVars.keySet();
   }
+
+  private void createQueryLog(String message) {
+    RollupLogger.Instance.Log(
+      message,
+      'Query: ' +
+      this.args.query +
+      '\nAccess level: ' +
+      (this.accessLevel == System.AccessLevel.SYSTEM_MODE ? 'System' : 'User'),
```

Comment:
`System.AccessLevel` is - confusingly - _not_ an enum, and the `toString()` implementation it has is ... not good ... so this ternary felt like the right compromise. It probably could be promoted to be part of the `RollupRepository` constructor though.